src/System.Linq/src/System/Linq/Buffer.cs

```diff
@@ -2,15 +2,14 @@
 // The .NET Foundation licenses this file to you under the MIT license.
 // See the LICENSE file in the project root for more information.
 
-using System;
 using System.Collections.Generic;
 
 namespace System.Linq
 {
     internal struct Buffer<TElement>
     {
-        internal TElement[] items;
-        internal int count;
+        internal readonly TElement[] _items;
+        internal readonly int _count;
```

Comment:
Jon Skeet wrote a [blog post](http://codeblog.jonskeet.uk/2014/07/16/micro-optimization-the-surprising-inefficiency-of-readonly-fields/) about performance of ValueTypes with readonly field. I wonder if the performance issues he found apply here.